### ISSUE TITLE
tsql: RAISEERROR breaks parsing

### ISSUE THREAD

**Issue by fmms** [sqlfluff/sqlfluff#3544]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have the following code:

```
CREATE PROCEDURE [UTIL].[createClusteredColumnstoreIndex] AS
BEGIN
    SET NOCOUNT ON;

    RAISERROR (
                'The specified table does not exist. Please enter @tableName in the following format: Schemaname.Tablename OR [Schemaname].[Tablename]'
                ,11
                ,- 1
                );

END
```

I tried to use sqlfluff to format it.

### Expected Behaviour

Code is formatted

### Observed Behaviour

sqlfluff cannot parse the file.

### How to reproduce

```
PS > cat .\sqlfluff-5.config
[sqlfluff]
dialect = tsql
templater = raw
PS > cat .\sqlfluff-5.sql
CREATE FUNCTION [dbo].[getItemList] (
     @list ItemList READONLY
)
RETURNS nvarchar(max)
AS

BEGIN
      DECLARE @str nvarchar(max) = ''

      SELECT @str = @str + [item] FROM (
        SELECT TOP (9999) [item]
        FROM @list
        ORDER BY [order]
      ) i

      RETURN @str
END
PS > sqlfluff parse .\sqlfluff-5.sql --config .\sqlfluff-5.config
```

### Dialect

tsql

### Version

```==== sqlfluff ====
sqlfluff:                1.0.0 python:                 3.10.4
implementation:        cpython verbosity:                   3
dialect:                  tsql templater:                 raw
rules:                                all
```

### Configuration

see above

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by fmms** [comment]
It seems to be something broken with the parser rather than no support at all. RAISERROR has been implemented at https://github.com/sqlfluff/sqlfluff/blob/a540ca1e639335d0a38e90dcf883daef5916c858/src/sqlfluff/dialects/dialect_tsql.py#L3603

**Comment by fmms** [comment]
Reasearching more, it might be an issue of https://github.com/sqlfluff/sqlfluff/blob/f447a604dffe04216f6414ecb0aa4d670e47383c/src/sqlfluff/dialects/dialect_ansi.py#L958 that it does not allow spaces in the `QualifiedNumericLiteralSegment`

**Comment by tunetheweb** [comment]
> Reasearching more, it might be an issue of
> 
> https://github.com/sqlfluff/sqlfluff/blob/f447a604dffe04216f6414ecb0aa4d670e47383c/src/sqlfluff/dialects/dialect_ansi.py#L958
> 
> that it does not allow spaces in the `QualifiedNumericLiteralSegment`

Yes that's correct. If tsql does indeed allow such a crazy syntax, then should override `QualifiedNumericLiteralSegment` in the tsql dialect to remove `allow_gaps=False,` line, though there's a risk this will confuse some expressions.

Alternatively would be to create a `SpacedQualifiedNumericLiteralSegment` and only use that in `RAISEERROR`.

Or other alternative, would be to accept that, while potentially valid, this is not best practice, and correct the SQL to remove the space and leave SQLFluff to highlight this (incorrectly as a parse error, rather than a rule error, but it's highlighting it none-the-less).

**Comment by fmms** [comment]
@tunetheweb not sure if the syntax is that crazy. I just used SQL fiddle to play a bit:

- Oracle accepts `SELECT 1 * - 1 FROM DUAL`
- Postgres accepts `SELECT 1 * - 1`
- MySQL accepts `SELECT 1 * - 1`
- SQL Server accepts `SELECT 1 * - 1`

So right now all databases seem to be fine with a space in between sign and number.
Which database does not accept that?

**Comment by barrywhart** [comment]
I would guess it's universally accepted, even though it's pretty unusual to use it. They almost have to, because unary minus is an operator, e.g. `-c`, I.e. the negative value of column `c`.

**Comment by tunetheweb** [comment]
Confirmed it's valid in BigQuery too, and also as per https://github.com/sqlfluff/sqlfluff/issues/3577#issuecomment-1179711157 multiple signs are also supported in at least some of these dialects.

#3581 raised to resolve this.
